### ISSUE TITLE
[Bug]: Proxy Meta cache SHOULD not carry any load information

### ISSUE THREAD

**Issue by xiaofan-luan** [milvus-io/milvus#37395]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

In Collection partial load feature, we introduced a dependency in proxy to check whether a field is loaded.

However, this is non necessary since querynode can always do the check itself.

The dependency itself brings a lot of trouble - 
1. Querycoord has to modify proxy cache on load/release, which is hard to manage and control.
2. When load happens,  it reported an error collection not load -> because it just need to check the collection meta info but right now it need to check if collection is loaded

2024-11-01T12:18:21.642480824+08:00 stdout F [2024/11/01 04:18:21.642 +00:00] [WARN] [proxy/task_scheduler.go:480] ["Failed to execute task: "] [traceID=a2498da8bc444264c53561450bfd3cfb] [error="call query coordinator LoadCollection: collection not fully loaded: collection not loaded[collection=453622615920019047]"]
2024-11-01T12:18:21.642566623+08:00 stdout F [2024/11/01 04:18:21.642 +00:00] [WARN] [proxy/impl.go:842] ["LoadCollection failed to WaitToFinish"] [traceID=a2498da8bc444264c53561450bfd3cfb] [role=proxy] [db=default] [collection=test_collection_2024_11_01_04_17_53_577009OylCGxDa] [refreshMode=true] [error="call query coordinator LoadCollection: collection not fully loaded: collection not loaded[collection=453622615920019047]"] [BeginTS=453623074413608965] [EndTS=453623074413608965]

detailed logs: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-37318/5/pipeline/

### Expected Behavior

1. proxy meta cache don't has any dependencies with querynode
2. even if it need any load information to be cached, and kind of load/release operation should not check whether it is loaded or not

**Comment by xiaofan-luan** [comment]
coul also related to https://github.com/milvus-io/milvus/issues/37144

**Comment by xiaofan-luan** [comment]
@congqixia 
how is the progress?

**Comment by congqixia** [comment]
After some attempt to move load list check logic out of proxy, I found two major blocker:

1.  search plan validation

The expression is parsed in proxy and marshaled into byte slice. Since we want the check happens as less as possible, the best fit is to check load list in delegator, which means the search plan has to be parsed/unmarshaled in delegator which may cost extra CPU time. And the parsed plan shall be traversed to check all the fields related are loaded. These change may cause performance issue in delegator search/query execution.

2. wildcard translate

see current impl:
https://github.com/milvus-io/milvus/blob/f42869c0ce2c5d669da333e64fc43789a711b551/internal/proxy/util.go#L1227-L1235 

the wildcard "*" is translated into field list according to schema or load list(if any) in proxy. If we wanna move the load list info into delegator, the field list could not be correct from proxy, say if field list is {"id": 100, "value": 101, "vector:"102"}, output field is ["*", "ID"], proxy cannot be shall the actual output field id, since field "value" may be not loaded. 

the possible workaround is to translate "*" into a position holder, say -1, and replace the actual load list in delegator.

To sum up the above, maybe for 2.5 release, we shall keep the current choice to keep load info cache in proxy and make the logic right. And design the execution routine as a whole to reduce the re-calculation.

Any comments? @xiaofan-luan

**Comment by yanliang567** [comment]
@congqixia any additonal following up actions?

**Comment by congqixia** [comment]
Should fixed by #42816
